pkg/rotator/rotator.go

```diff
@@ -496,22 +498,23 @@ func (m *mapper) Map(object handler.MapObject) []reconcile.Request {
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func addController(mgr manager.Manager, r *ReconcileWH) error {
-	vwh := &unstructured.Unstructured{}
-	vwh.SetGroupVersionKind(vwhGVK)
+func addController(mgr manager.Manager, r *ReconcileWH, webhookInfos []WebhookInfo) error {
 	crd := &unstructured.Unstructured{}
 	crd.SetGroupVersionKind(crdGVK)
 	// Create a new controller
-	err := ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Secret{}).
-		Watches(&source.Kind{Type: vwh}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &mapper{
+	builder := ctrl.NewControllerManagedBy(mgr).For(&corev1.Secret{})
+	for _, webhookInfo := range webhookInfos {
+		wh := &unstructured.Unstructured{}
+		wh.SetGroupVersionKind(webhookInfo.GVK)
+		builder = builder.Watches(&source.Kind{Type: wh}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &mapper{
 			secretKey: r.secretKey,
-			vwhKey:    r.vwhKey,
-		}}).
-		Watches(&source.Kind{Type: crd}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &crdMapper{
-			secretKey: r.secretKey,
-			crdNames:  r.crdNames,
-		}}).Complete(r)
+			vwhKey:    webhookInfo.Name,
+		}})
+	}
+	err := builder.Watches(&source.Kind{Type: crd}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &crdMapper{
+		secretKey: r.secretKey,
+		crdNames:  r.crdNames,
```

Comment:
If you generalize to CRDs you don't need this line, right?

Reply:
Removed